### PR TITLE
HPCC-21296 Create Cluster Processes Widget

### PR DESCRIPTION
**Pull request by miguelvazq**
Fixes: HPCC-21296

Signed-off by: Miguel Vazquez <miguel.vazquez@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [ ] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by miguelvazq** [comment]
@GordonSmith Please review. This is dependent on HPCC-21295

File: esp/src/eclwatch/ClusterProcessesQueryWidget.js [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+                        label: this.i18n.Directory,
+                        sortable: false,
+                        width: 200
+                    },
+                    LogDirectory: {
+                        label: this.i18n.LogDirectory,
+                        sortable: false,
+                        width: 200
+                    }
+                }
```
Comment (GordonSmith) [inline esp/src/eclwatch/ClusterProcessesQueryWidget.js]: Should there be one column without width?

**Review by GordonSmith — CHANGES_REQUESTED** [review]
Minor comments.

File: esp/src/eclwatch/ClusterProcessesQueryWidget.js (hunk 1 above)
Comment (miguelvazq) [reply esp/src/eclwatch/ClusterProcessesQueryWidget.js]: Removing a column width causes issues within the grid.

**Comment by miguelvazq** [comment]
Issue in smoke test will be resolved once #12624 is resolved.

File: esp/src/eclwatch/ClusterProcessesQueryWidget.js [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+            var selections = this.grid.getSelected();
+            var firstTab = null;
+            for (var i = 0; i < selections.length; ++i) {
+                WsTopology.TpGetComponentFile({
+                    request: {
+                        FileType: "cfg",
+                        CompType: selections[i].Component,
+                        CompName: selections[i].Name,
+                        Directory: selections[i].Directory,
+                        OsType: selections[i].OS,
```
Comment (GordonSmith) [inline esp/src/eclwatch/ClusterProcessesQueryWidget.js]: Remove trailing comma

File: esp/src/eclwatch/ClusterProcessesQueryWidget.js [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+            }
+        },
+
+        _onRefresh: function () {
+            this.refreshGrid();
+        },
+
+        refreshGrid: function () {
+            this.grid.set("query", {
+                Type: "ROOT",
```
Comment (GordonSmith) [inline esp/src/eclwatch/ClusterProcessesQueryWidget.js]: Remove trailing comma

**Review by GordonSmith — CHANGES_REQUESTED** [review]
@miguelvazq can you run a local lint as well as removing those trailing commas?

**Comment by miguelvazq** [comment]
@GordonSmith Rebased and updated per comments.

**Pull request merged**